src/Compilers/VisualBasic/Portable/VisualBasicCompilationOptions.vb

```diff
@@ -1199,5 +1199,54 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 referencesSupersedeLowerVersions:=False)
 
         End Sub
+
+        Protected Overrides Function CommonWithModuleName(moduleName As String) As CompilationOptions
+            Return WithModuleName(moduleName)
+        End Function
+
+        Protected Overrides Function CommonWithMainTypeName(mainTypeName As String) As CompilationOptions
+            Return WithMainTypeName(mainTypeName)
+        End Function
+
+        Protected Overrides Function CommonWithScriptClassName(scriptClassName As String) As CompilationOptions
+            Return WithScriptClassName(scriptClassName)
+        End Function
+
+        Protected Overrides Function CommonWithCryptoKeyContainer(cryptoKeyContainer As String) As CompilationOptions
+            Return WithCryptoKeyContainer(cryptoKeyContainer)
+        End Function
+
+        Protected Overrides Function CommonWithCryptoKeyFile(cryptoKeyFile As String) As CompilationOptions
+            Return WithCryptoKeyFile(cryptoKeyFile)
+        End Function
+
+        Protected Overrides Function CommonWithCryptoPublicKey(cryptoPublicKey As ImmutableArray(Of Byte)) As CompilationOptions
+            Return WithCryptoPublicKey(cryptoPublicKey)
+        End Function
+
+        Protected Overrides Function CommonWithDelaySign(delaySign As Boolean?) As CompilationOptions
+            Return WithDelaySign(delaySign)
+        End Function
+
+        Protected Overrides Function CommonWithCheckOverflow(checkOverflow As Boolean) As CompilationOptions
+            Return WithOverflowChecks(checkOverflow)
+        End Function
+
+        ''' <summary>
+        ''' Creates a new VisualBasicCompilationOptions instance with a different warning level specified.
+        ''' </summary>
+        ''' <param name="value">The warning level setting. </param>        
+        ''' <returns>A new instance of VisualBasicCompilationOptions, if the warning level is different; otherwise current instance.</returns>        
+        Public Shadows Function WithWarningLevel(value As Integer) As VisualBasicCompilationOptions
+            If value = Me.WarningLevel Then
+                Return Me
+            End If
+
+            Return New VisualBasicCompilationOptions(Me) With {.WarningLevel = value}
+        End Function
```

Comment:
We probably shouldn't add this member to VB, it is not one of the options affecting it. We probably shouldn't add it to the common options as well.